src/main/java/edu/cornell/kfs/fp/document/web/struts/RecurringDisbursementVoucherForm.java

```diff
@@ -1,54 +1,105 @@
-	private transient RecurringDisbursementVoucherDocumentService recurringDisbursementVoucherDocumentService;
-	
-	@Override
-	public String getDocTypeName() {
-		return CuFPConstants.RecurringDisbursementVoucherDocumentConstants.RECURRING_DV_DOCUMENT_TYPE_NAME;
-	}
-	
-	@Override
+
+    private static final long serialVersionUID = 7035540080454973823L;
+    private static final org.apache.log4j.Logger LOG = org.apache.log4j.Logger.getLogger(RecurringDisbursementVoucherForm.class);
+
+    private transient List<RecurringDisbursementVoucherPDPStatus> pdpStatuses;
+    private transient RecurringDisbursementVoucherDocumentService recurringDisbursementVoucherDocumentService;
+    private transient RecurringDisbursementVoucherPaymentMaintenanceService recurringDisbursementVoucherPaymentMaintenanceService;
+
+    @Override
+    public String getDocTypeName() {
+        return CuFPConstants.RecurringDisbursementVoucherDocumentConstants.RECURRING_DV_DOCUMENT_TYPE_NAME;
+    }
+
+    @Override
     protected String getDefaultDocumentTypeName() {
         return CuFPConstants.RecurringDisbursementVoucherDocumentConstants.RECURRING_DV_DOCUMENT_TYPE_NAME;
     }
-	
-	@Override
-	public String getTripUrl() {
-		return StringUtils.EMPTY;
-	}
-	
-	@Override
-	public String getTripID() {
-		return StringUtils.EMPTY;
-	}
-	
-	public boolean isRecurringDVDetailsDefaultOpen() {
-		return !((RecurringDisbursementVoucherDocument)this.getDocument()).getRecurringDisbursementVoucherDetails().isEmpty();
-	}
-	
-	public int getRecurringDVDetailsSize() {
-	    return ((RecurringDisbursementVoucherDocument)this.getDocument()).getRecurringDisbursementVoucherDetails().size();
-	}
-	
-	public boolean isPreDisbursementProcessorTabDefaultOpen() {
-	    return !getPdpStatuses().isEmpty();
-	}
+
+    @Override
+    public String getTripUrl() {
+        return StringUtils.EMPTY;
+    }
+
+    @Override
+    public String getTripID() {
+        return StringUtils.EMPTY;
+    }
+
+    public boolean isCancelPDPPaymentsActionAvailable() {
+        if (getRecurringDisbursementVoucherPaymentMaintenanceService().hasCancelPermission(GlobalVariables.getUserSession().getPerson())) {
+            return arePaymentDetailsCancelAble() || areDVsCancelable();
+        }
+        return false;
+    }
+    
+    private boolean arePaymentDetailsCancelAble() {
+        Collection<PaymentDetail> details = getRecurringDisbursementVoucherDocumentService().findPaymentDetailsFromRecurringDisbursementVoucher((RecurringDisbursementVoucherDocument)getDocument());
+        for (PaymentDetail detail : details) {
+            String paymentDetailStatus = detail.getPaymentGroup().getPaymentStatusCode();
+            
+            if (getRecurringDisbursementVoucherDocumentService().isPaymentCancelableByLoggedInUser(paymentDetailStatus)) {
+                return true;
+            }
+        }
+        return false;
+    }
+    
+    private boolean areDVsCancelable() {
```

Comment:
I would change this name to areAnyDVsCancelable since it seems you are looking for at least one DV to cancelable but you don't need all of them to be cancelable